src/main/java/org/apache/commons/text/WordUtils.java

```diff
@@ -414,18 +414,29 @@ public static String capitalize(final String str, final char... delimiters) {
         if (StringUtils.isEmpty(str) || delimLen == 0) {
             return str;
         }
-        final char[] buffer = str.toCharArray();
+        int strLen = str.length();
+        int [] newCodePoints = new int[strLen];
+        int outOffset = 0;
+
         boolean capitalizeNext = true;
-        for (int i = 0; i < buffer.length; i++) {
-            final char ch = buffer[i];
-            if (isDelimiter(ch, delimiters)) {
+        for (int index = 0; index < str.length();) {
```

Comment:
`str.length();` could be replaced with `strLen` initialized above (just from a performance standpoint). Note, this is extra nit-picky and need not block the PR.